frontend/src/lib/integrations/IntegrationScopesWarning.tsx

```diff
@@ -0,0 +1,64 @@
+import api from 'lib/api'
+import { LemonBanner } from 'lib/lemon-ui/LemonBanner'
+import { Link } from 'lib/lemon-ui/Link'
+
+import { HogFunctionInputSchemaType, IntegrationType } from '~/types'
+
+export function IntegrationScopesWarning({
+    integration,
+    schema,
+}: {
+    integration: IntegrationType
+    schema?: HogFunctionInputSchemaType
+}): JSX.Element {
+    const getScopes = (): string[] => {
+        const scopes: any[] = []
+        const possibleScopeLocation = [integration.config.scope, integration.config.scopes]
+
+        possibleScopeLocation.map((scope) => {
+            if (typeof scope === 'string') {
+                scopes.push(scope.split(' '))
+                scopes.push(scope.split(','))
+            }
+            if (typeof scope === 'object') {
+                scopes.push(scope)
+            }
+        })
+        return scopes
+            .filter((scope: any) => typeof scope === 'object')
+            .reduce((a, b) => (a.length > b.length ? a : b), [])
+    }
+
+    const requiredScopes = schema?.requiredScopes?.split(' ') || []
+    const missingScopes = requiredScopes.filter((scope: string) => !getScopes().includes(scope))
```

Comment:
Small note - this is a non-trivial amount of code run every render. Can be moved into a `useMemo` so that it only runs when the integration object changes